### DISCUSSION TITLE
Compatibility with `@ionic/vue-router`

### DISCUSSION THREAD

**Discussion by reslear** [posva/unplugin-vue-router#278] [Questions and Help]
is there any chance of using with the [@ionic/vue-router](https://github.com/ionic-team/ionic-framework/blob/main/packages/vue-router/src/router.ts) ?

**Comment by posva** [comment]
Probably but I haven't used ionic yet. The types part might need extra work. Feel free to report your advances on that here for other users!

**Comment by ozgurg** [comment]
I attempted to use it, but I couldn't manage it. Replacing `@ionic/vue-router` with `vue-router/auto` and `<ion-router-outlet />` with `<router-view />` breaks Ionic's transitions and other functionalities it provides.

**Comment by reslear** [comment]
Base on nuxt module by @danielroe, router implementation can be complicated   
https://github.com/nuxt-modules/ionic/blob/main/src/runtime/plugins/router.ts

**Comment by fcl999** [comment]
By using the following method, the first parameter of useIonRouter can receive type hints and route generation.

router/index.ts
```ts
/**
 * router/index.ts
 *
 * Automatic routes for `./src/pages/*.vue`
 */

// Composables
import { routes } from "vue-router/auto-routes";
import { createRouter, createWebHistory } from "@ionic/vue-router";

const router = createRouter({
  history: createWebHistory(import.meta.env.BASE_URL),
  routes,
});
```

router.d.ts

```ts
/**
 * router.d.ts
 */
import { type AnimationBuilder } from "@ionic/vue";
import { type RouteLocationRaw } from "vue-router/auto";
import type { RouteAction } from "@ionic/vue-router/dist/types/types";
import type { RouteDirection } from "@ionic/vue/dist/types/hooks/router";

export interface UseIonRouterResultUseRouteLocationRaw {
  canGoBack: (deep?: number) => boolean;
  push: (
    location: RouteLocationRaw,
    routerAnimation?: AnimationBuilder,
  ) => void;
  replace: (
    location: RouteLocationRaw,
    routerAnimation?: AnimationBuilder,
  ) => void;
  back: (routerAnimation?: AnimationBuilder) => void;
  forward: (routerAnimation?: AnimationBuilder) => void;
  navigate: (
    location: RouteLocationRaw,
    routerDirection?: RouteDirection,
    routerAction?: RouteAction,
    routerAnimation?: AnimationBuilder,
  ) => void;
}

declare module "@ionic/vue" {
  export declare const useIonRouter: () => UseIonRouterResultUseRouteLocationRaw;
}
```

**Comment by reslear** [comment]
so i tested works well, https://github.com/reslear/vitesse-lite-ionic-layouts
but without layouts, need find solution :)
